nodes/default_node.go

```diff
@@ -38,8 +38,12 @@ type DefaultNode struct {
 	// OverwriteNode stores the interface used to overwrite methods defined
 	// for DefaultNode, so that particular nodes can provide custom implementations.
 	OverwriteNode NodeOverwrites
+	// List of links that reference the node.
+	Links []links.Link
 	// List of link endpoints that are connected to the node.
 	Endpoints []links.Endpoint
+	// State of the node
+	State string
```

Comment:
I think an easy and dirty way to create a state enum and not deal with circular deps much is to create a package like nodes/state and put the enum there

Reply:
done in [25d99e0](https://github.com/srl-labs/containerlab/pull/1507/commits/25d99e0795ab87507baf4d7a68fb6cf8166f88f7)